src/main/java/org/dinosaur/foodbowl/domain/auth/application/AuthService.java

```diff
@@ -0,0 +1,53 @@
+package org.dinosaur.foodbowl.domain.auth.application;
+
+import lombok.RequiredArgsConstructor;
+import org.dinosaur.foodbowl.domain.auth.apple.AppleOAuthUserProvider;
+import org.dinosaur.foodbowl.domain.auth.dto.FoodbowlTokenDto;
+import org.dinosaur.foodbowl.domain.auth.dto.request.AppleLoginRequestDto;
+import org.dinosaur.foodbowl.domain.auth.dto.response.ApplePlatformUserResponseDto;
+import org.dinosaur.foodbowl.domain.member.repository.MemberRepository;
+import org.dinosaur.foodbowl.global.config.security.jwt.JwtTokenProvider;
+import org.dinosaur.foodbowl.global.exception.FoodbowlException;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.stereotype.Service;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.dinosaur.foodbowl.domain.member.entity.Member.SocialType;
+import static org.dinosaur.foodbowl.domain.member.entity.Role.RoleType;
+import static org.dinosaur.foodbowl.global.exception.ErrorStatus.APPLE_NOT_REGISTER;
+
+@RequiredArgsConstructor
+@Service
+public class AuthService {
+
+    private final MemberRepository memberRepository;
+    private final JwtTokenProvider jwtTokenProvider;
+    private final AppleOAuthUserProvider appleOAuthUserProvider;
+    private final RedisTemplate<String, Object> redisTemplate;
+
+    public FoodbowlTokenDto appleLogin(AppleLoginRequestDto appleLoginRequestDto) {
+        ApplePlatformUserResponseDto applePlatformUserResponseDto =
+                appleOAuthUserProvider.extractApplePlatformUser(appleLoginRequestDto.getAppleToken());
+        String socialId = applePlatformUserResponseDto.getSocialId();
+
+        return memberRepository.findBySocialTypeAndSocialId(SocialType.APPLE, socialId)
+                .map(member -> {
+                    String accessToken = jwtTokenProvider.createAccessToken(member.getId(), RoleType.ROLE_회원);
+                    String refreshToken = jwtTokenProvider.createRefreshToken(member.getId());
+                    registerRefreshToken(member.getId(), refreshToken);
+                    return new FoodbowlTokenDto(accessToken, refreshToken);
+                })
+                .orElseThrow(() -> new FoodbowlException(APPLE_NOT_REGISTER));
+    }
```

Comment:
return 하는 과정에서 많은 일을 하고 있는 것 같은데 분리해보는건 어떨까요?
```suggestion
public FoodbowlTokenDto appleLogin(AppleLoginRequestDto appleLoginRequestDto) {
       
    ...
    Member member = memberRepository.findBySocialTypeAndSocialId(SocialType.APPLE, socialId)
            .orElseThrow(() -> new FoodbowlException(APPLE_NOT_REGISTER));
    
    return generateToken(member.getId(), RoleType.ROLE_회원);
}
    
private FoodbowlTokenDto generateToken(Long id, RoleType roleType) {
      String accessToken = jwtTokenProvider.createAccessToken(id, roleType);
      String refreshToken = jwtTokenProvider.createRefreshToken(id);
      registerRefreshToken(id, refreshToken);
      return new FoodbowlTokenDto(accessToken, refreshToken);
}
```

요런식으로 해볼 수도 있을 것 같아요 😀

Reply:
너무 좋은데요 😆👍